File: 77-combinations/77-combinations.py
class Solution(object):
    def combine(self, n, k):
        """
        :type n: int
        :type k: int
        :rtype: List[List[int]]
        """
        res = []
        final = []
        def find(i):
            if len(res) == k:
                final.append(res[:])
                return
            for elem in range(i,n+1):
                res.append(elem)
                find(elem+1)
                res.pop()
        find(1)
        return final
        

File: 77-combinations/77-combinations.java
class Solution {
    private LinkedList<Integer> temp = new LinkedList<>();
    private List<List<Integer>> res = new LinkedList<>();
    private int start = 1;
    public List<List<Integer>> combine(int n, int k) {
        if( k == 0){
            res.add((List) temp.clone());
            return res;
        }
        
        for(int i = start; i < n + 1; i++){
            temp.addLast(i);
            start = i + 1;
            combine(n, k - 1);
            temp.pollLast();
        }
        return res;
    }
}